### PR TITLE
Add pre-commit hook to check release note bullet point stlye

### LINKED ISSUE #36173: Pre-commit check for release note bullet point style
**Is your feature request related to a problem? Please describe.**
Release notes must now use `-` as the bullet point style (instead of `*`) due to warnings created when mixing bullet point styles during note amalgamating, creating the main release note pages. The error is confusing `WARNING: Bullet list ends without a blank line; unexpected unindent.`.

**Describe the solution you'd like**
Add a simple pre-commit check which looks to see if a release note has been added, and if so checks that the first character of each line is a `-`. It Should also cope with indented bullet point lists:
```
- Thing one

	1. sub thing 1
	2. sub thing 2
```

Maybe the most bulletproof method would be to add a `-` bullet point on the line above and below the note, and ask sphinx to check for warnings.

### PR DESCRIPTION
**Pull request by jhaigh0**
**Description of work**

Add a pre-commit hook to check that new release notes use the ``-`` style of bullet points instead of ``*`` or ``+``.
Bullet points part of a nested list aren't checked.

To do this I've added a pre-commit to [this repo](https://github.com/mantidproject/pre-commit-hooks)

(also the pre commit bot has found a fix on this pr, I guess the pr pre-commit checks are run on all files?)

**Purpose of work**
When release notes are combined into the main files, conflicting bullet point styles cause a confusing error. Furthermore, this error won't occur if you use the wrong style but are the first to make a note for that section, meaning the problem only gets caught later.

**To test:**

 - Make some release notes which test that
   - Good formatting is left alone
   - Bad bullet point styles are changed (when not in a nested list)
   - The correct files are picked up
   - Really anything you can think of 

Fixes #36173 

---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

**Review by robertapplin — DISMISSED** [review]
This works really well, and will prevent a lot of problems when building our release notes with sphinx. This will mean fewer nightly failures due to sphinx warnings, thanks!

**Review by cailafinn — DISMISSED** [review]
I'm not overly familiar with how pre-commit works, but is there a reason a new repo was created for this and rather than using the existing [pre-commit-hooks repo](https://github.com/mantidproject/pre-commit-hooks), which is already within the organisation?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -51,3 +51,9 @@ repos:
     hooks:
       - id: black
         exclude: scripts/templates/reference/|Testing/Tools/cxxtest
+
+  - repo: https://github.com/jhaigh0/mantid-pre-commit
```
Comment (peterfpeterson) [inline .pre-commit-config.yaml:55]: Shouldn't the hook be in https://github.com/mantidproject/pre-commit-hooks.git ?

**Comment by peterfpeterson** [comment]
I only got through fixing issues in the dev-docs site, but #36288 might flag this issue in a more robust way.

**Comment by jhaigh0** [comment]
I have added a PR to the mantid project pre commit hooks repo. So once that is through I can update this PR to use that instead.

https://github.com/mantidproject/pre-commit-hooks/pull/10

**Comment by ConorMFinn** [comment]
- https://github.com/mantidproject/pre-commit-hooks/pull/10 is now merged, so this is ready for another look @robertapplin

**Review by peterfpeterson — APPROVED** [review]

**Pull request merged**